diff --git a/make/autoconf/platform.m4 b/make/autoconf/platform.m4
index babb24a9b..7b770f0e3 100644
--- a/make/autoconf/platform.m4
+++ b/make/autoconf/platform.m4
@@ -206,6 +206,10 @@ AC_DEFUN([PLATFORM_EXTRACT_VARS_FROM_OS],
       VAR_OS=macosx
       VAR_OS_TYPE=unix
       ;;
+    *ios*)
+      VAR_OS=macosx
+      VAR_OS_TYPE=unix
+      ;;
     *bsd*)
       VAR_OS=bsd
       VAR_OS_TYPE=unix