diff --git a/src/hotspot/os/bsd/os_bsd.cpp b/src/hotspot/os/bsd/os_bsd.cpp
index faf9d9f64..8828e855b 100644
--- a/src/hotspot/os/bsd/os_bsd.cpp
+++ b/src/hotspot/os/bsd/os_bsd.cpp
@@ -757,26 +757,6 @@ void os::Bsd::clock_init() {
 
 
 
-#ifdef __APPLE__
-static bool rwxChecked, rwxAvailable;
-#endif
-bool os::Bsd::isRWXJITAvailable() {
-#ifdef __APPLE__
-  if (!rwxChecked) {
-    rwxChecked = true;
-    const int flags = MAP_PRIVATE | MAP_NORESERVE | MAP_ANONYMOUS | MAP_JIT;
-    char* addr = (char*)::mmap(0, getpagesize(), PROT_NONE, flags, -1, 0);
-    rwxAvailable = addr != MAP_FAILED;
-    if (rwxAvailable) {
-      ::munmap(addr, getpagesize());
-    }
-  }
-  return rwxAvailable;
-#else
-  return true;
-#endif
-}
-
 #ifdef __APPLE__
 
 jlong os::javaTimeNanos() {
@@ -1576,7 +1576,7 @@ bool os::pd_commit_memory(char* addr, size_t size, bool exec) {
 #elif defined(__APPLE__)
   if (exec) {
     // Do not replace MAP_JIT mappings, see JDK-8234930
-    if (::mprotect(addr, size, prot) == 0) {
+    if (true || ::mprotect(addr, size, prot) == 0) {
       return true;
     }
   } else {
diff --git a/src/hotspot/os/bsd/os_bsd.hpp b/src/hotspot/os/bsd/os_bsd.hpp
index 320709429..1887a6c2b 100644
--- a/src/hotspot/os/bsd/os_bsd.hpp
+++ b/src/hotspot/os/bsd/os_bsd.hpp
@@ -62,6 +62,11 @@ class os::Bsd {
   static bool hugetlbfs_sanity_check(bool warn, size_t page_size);
 
  public:
+  static bool isRWXJITAvailable() {
+    // For iOS, we don't have RWX JIT, so return false
+    return false;
+  }
+
   static void init_thread_fpu_state();
   static pthread_t main_thread(void)                                { return _main_thread; }